TaskWithClass/MyString.cpp

```diff
@@ -0,0 +1,254 @@
+#include "pch.h"
+#include "MyString.h"
+
+const char* String::c_str() const noexcept
+{
+	return m_str;
+}
+
+size_t String::length() const
+{
+	return m_len;
+}
+
+size_t String::size() const
+{
+	return m_len;
+}
+
+size_t String::capacity() const noexcept
+{
+	return m_capacity;
+}
+
+void String::resize(size_t n, char c)
+{
+	if (m_len > n)
+	{
+		m_str[n] = '\0';
+		this->m_len = n;
+	}
+	else if (m_len < n)
+	{
+		char* mem = new char[n + 1];
+		memcpy(mem, m_str, m_len);
+		for (size_t i = m_len; i <= n - 1; i++)
+		{
+			mem[i] = c;
+		}
+		mem[n] = '\0';
+		delete[] m_str;
+		this->m_str = mem;
+		if (c != '\0')
+		{
+			this->m_len = n;
+		}
+		this->m_capacity = n + 1;
+	}
+}
+
+String& String::append(const String& str)
+{
+	if (str.length() == 0)
+	{
+		return *this;
+	}
+	size_t len = this->length() + str.length();
```

Comment:
If a new length is the same, i.e. we add an empty string then no operation is needed. Please also keep in mind if capacity is big then you don't need to delete & allocate.